h/db/mixins.py

```diff
@@ -5,13 +5,16 @@
 import sqlalchemy as sa
 
 
-class Timestamps:
+class CreatedMixin:
```

Comment:
CreatedMixin makes sense for "immutable" rows like the moderation log.

I don't think it makes sense to have a mixing just for updated.

Reply:
This change would make more sense on the model PR but leaving it here to avoid a chain of rebases.

It's on its own commit anyway.

Reply:
Oh, do we wanna use SQLAlchemy's dataclasses integration for the new table?

This is now possible in h. For an example see [the new `task_done` table](https://github.com/hypothesis/h/blob/c0805caa6638ca21ff951cb3ecbff4b23a1f3d81/h/models/task_done.py#L13-L21) and [`mixins_dataclasses.py`](https://github.com/hypothesis/h/blob/c0805caa6638ca21ff951cb3ecbff4b23a1f3d81/h/db/mixins_dataclasses.py#L13-L34)

Reply:
Oh I did use the dataclass mixin on the model here:

https://github.com/hypothesis/h/pull/9478/files#diff-1b00ee8fcdfb509398d2b2cd202b3407c84f46f64145a0fc27006b25cd7edf82R15

But I didn't include this mixin on the mixins_dataclasses.py module. 

I'm not 100% sure but I reckon as long as the model has the dataclass mixin the rest of the mixins don't need it? That would be useful to avoid to migrate all the classes using the mixing in one go.

Reply:
> I'm not 100% sure but I reckon as long as the model has the dataclass mixin the rest of the mixins don't need it?

No I think that every mixin class needs to inherit from `MappedAsDataclass` as well. That's what we've been doing in Via. See here in the SQLAlchemy docs: https://docs.sqlalchemy.org/en/20/orm/dataclasses.html#using-mixins-and-abstract-superclasses

Reply:
:thinking: The docs are pretty clear about it but the created column did appear on the DB.

I'll double check that.

Reply:
Ah, so while alembic and sqlalchemy itself are fine with missing the dataclass mixin on the other mixin it is required for typechecking:

> Python type checkers which support PEP 681 will otherwise not consider attributes from non-dataclass mixins to be part of the dataclass.

but we don't actually access `created` directly so we haven't got a warning for this. 

I'll fix it.